### PR TITLE
Fix tab control border in high dpi

### PR DESCRIPTION
**Pull request by NikolayXHD**
Fixes the issue discussed in https://github.com/gitextensions/gitextensions/pull/8732#issuecomment-757439317

## Proposed changes

Let system draw tab control. To remain compatible with dark theme use Win32 api overrides.

Unfortunately tab control now flickers on mouse move. `.SetStyle(DobuleBuffer, true)` would not help,
because double buffer only works with `UserPaint | AllPaintingInWmPaint`

### Before
image

### After
image

## Test environment(s) 

- Windows 10, dpi 150%

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by ivangrek — COMMENTED** [review]
image

Different colors

**Review by ivangrek — APPROVED** [review]

**Review by gerhardol — COMMENTED** [review]
+1
looks fine to me. The implementation seems easier to me too - the best practices for overriding is not my speciality

indent issue in Win32ThemeHooks:283

**Comment by NikolayXHD** [comment]
> indent issue in Win32ThemeHooks:283

Fixed

**Comment by NikolayXHD** [comment]
> image
> 
> Different colors

Fixed

File: GitUI/Theming/Renderers/TabRenderer.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        private static void RenderPane(Context ctx, Rectangle prect)
+        {
+            var borderPen = GetBorderPen();
+            ctx.Graphics.DrawRectangle(borderPen, prect.Inclusive());
+        }
+
+        private static void RenderTab(Parts partId, States stateId, Rectangle prect, Context ctx)
+        {
+            Brush backBrush = GetTabBackgroundBrush(stateId);
```
Comment (RussKie) [inline GitUI/Theming/Renderers/TabRenderer.cs:58]: Doesn't it need to be disposed of?

**Comment by RussKie** [comment]
The tab highlighting looks fine in the light version, acceptable in the dark version but doesn't look good in the darksilver version:

* light
image

* darksilver
image

Any pointers to what should be tweaked?

File: GitUI/Theming/Renderers/TabRenderer.cs (hunk 1 above)
Comment (NikolayXHD) [reply GitUI/Theming/Renderers/TabRenderer.cs:58]: Here we use on of SystemBrushes, since we do not create it, we should not dispose.

**Comment by mdonatas** [comment]
before
tabs-not-flickering

after
tabs-flickering

This change has caused a very undesirable flickering. Tested at 200% scaling

**Pull request merged**